cudamat/cudamat/libcudamat.so: cannot open shared object file: No such file or directory

copying cudamat.egg-info/pbr.json -> build/bdist.linux-x86_64/egg/EGG-INFO
copying cudamat.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
writing build/bdist.linux-x86_64/egg/EGG-INFO/native_libs.txt
zip_safe flag not set; analyzing archive contents...
cudamat.cudamat: module references __file__
cudamat.learn: module references __file__
creating dist
creating 'dist/cudamat-0.3-py2.7-linux-x86_64.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing cudamat-0.3-py2.7-linux-x86_64.egg
removing '/usr/local/lib/python2.7/dist-packages/cudamat-0.3-py2.7-linux-x86_64.egg' (and everything under it)
creating /usr/local/lib/python2.7/dist-packages/cudamat-0.3-py2.7-linux-x86_64.egg
Extracting cudamat-0.3-py2.7-linux-x86_64.egg to /usr/local/lib/python2.7/dist-packages
cudamat 0.3 is already the active version in easy-install.pth

Installed /usr/local/lib/python2.7/dist-packages/cudamat-0.3-py2.7-linux-x86_64.egg
Processing dependencies for cudamat==0.3
Finished processing dependencies for cudamat==0.3
ubgpu@ubgpu:~/github/cudamat$ 
ubgpu@ubgpu:~/github/cudamat$ 
ubgpu@ubgpu:~/github/cudamat$ 
ubgpu@ubgpu:~/github/cudamat$ nosetests
EEE
======================================================================
ERROR: Failure: OSError (/home/ubgpu/github/cudamat/cudamat/libcudamat.so: cannot open shared object file: No such file or directory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/cudamat/cudamat/__init__.py", line 1, in <module>
    from .cudamat import *
  File "/home/ubgpu/github/cudamat/cudamat/cudamat.py", line 14, in <module>
    os.path.dirname(__file__) or os.path.curdir, 'libcudamat' + sysconfig.get_config_var('SO')))
  File "/usr/lib/python2.7/ctypes/__init__.py", line 443, in LoadLibrary
    return self._dlltype(name)
  File "/usr/lib/python2.7/ctypes/__init__.py", line 365, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /home/ubgpu/github/cudamat/cudamat/libcudamat.so: cannot open shared object file: No such file or directory

======================================================================
ERROR: Failure: OSError (/home/ubgpu/github/cudamat/cudamat/libcudamat.so: cannot open shared object file: No such file or directory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/cudamat/test/test_cudamat.py", line 3, in <module>
    import cudamat as cm
  File "/home/ubgpu/github/cudamat/cudamat/__init__.py", line 1, in <module>
    from .cudamat import *
  File "/home/ubgpu/github/cudamat/cudamat/cudamat.py", line 14, in <module>
    os.path.dirname(__file__) or os.path.curdir, 'libcudamat' + sysconfig.get_config_var('SO')))
  File "/usr/lib/python2.7/ctypes/__init__.py", line 443, in LoadLibrary
    return self._dlltype(name)
  File "/usr/lib/python2.7/ctypes/__init__.py", line 365, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /home/ubgpu/github/cudamat/cudamat/libcudamat.so: cannot open shared object file: No such file or directory

======================================================================
ERROR: Failure: OSError (/home/ubgpu/github/cudamat/cudamat/libcudamat.so: cannot open shared object file: No such file or directory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/cudamat/test/test_learn.py", line 4, in <module>
    import cudamat as cm
  File "/home/ubgpu/github/cudamat/cudamat/__init__.py", line 1, in <module>
    from .cudamat import *
  File "/home/ubgpu/github/cudamat/cudamat/cudamat.py", line 14, in <module>
    os.path.dirname(__file__) or os.path.curdir, 'libcudamat' + sysconfig.get_config_var('SO')))
  File "/usr/lib/python2.7/ctypes/__init__.py", line 443, in LoadLibrary
    return self._dlltype(name)
  File "/usr/lib/python2.7/ctypes/__init__.py", line 365, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /home/ubgpu/github/cudamat/cudamat/libcudamat.so: cannot open shared object file: No such file or directory

----------------------------------------------------------------------
Ran 3 tests in 0.102s

FAILED (errors=3)
ubgpu@ubgpu:~/github/cudamat$

not too bad;)

ubgpu@ubgpu:~/github$ nosetests cudamat/ .
EEEEEEEEEEUsing gpu device 0: GeForce GTX 970
EEEEEEEEEEEEEE........................................F.......
======================================================================
ERROR: Failure: OSError (/home/ubgpu/github/cudamat/cudamat/libcudamat.so: cannot open shared object file: No such file or directory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/cudamat/cudamat/__init__.py", line 1, in <module>
    from .cudamat import *
  File "/home/ubgpu/github/cudamat/cudamat/cudamat.py", line 14, in <module>
    os.path.dirname(__file__) or os.path.curdir, 'libcudamat' + sysconfig.get_config_var('SO')))
  File "/usr/lib/python2.7/ctypes/__init__.py", line 443, in LoadLibrary
    return self._dlltype(name)
  File "/usr/lib/python2.7/ctypes/__init__.py", line 365, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /home/ubgpu/github/cudamat/cudamat/libcudamat.so: cannot open shared object file: No such file or directory

======================================================================
ERROR: Failure: OSError (/home/ubgpu/github/cudamat/cudamat/libcudamat.so: cannot open shared object file: No such file or directory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/cudamat/test/test_cudamat.py", line 3, in <module>
    import cudamat as cm
  File "/home/ubgpu/github/cudamat/cudamat/__init__.py", line 1, in <module>
    from .cudamat import *
  File "/home/ubgpu/github/cudamat/cudamat/cudamat.py", line 14, in <module>
    os.path.dirname(__file__) or os.path.curdir, 'libcudamat' + sysconfig.get_config_var('SO')))
  File "/usr/lib/python2.7/ctypes/__init__.py", line 443, in LoadLibrary
    return self._dlltype(name)
  File "/usr/lib/python2.7/ctypes/__init__.py", line 365, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /home/ubgpu/github/cudamat/cudamat/libcudamat.so: cannot open shared object file: No such file or directory

======================================================================
ERROR: Failure: OSError (/home/ubgpu/github/cudamat/cudamat/libcudamat.so: cannot open shared object file: No such file or directory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/cudamat/test/test_learn.py", line 4, in <module>
    import cudamat as cm
  File "/home/ubgpu/github/cudamat/cudamat/__init__.py", line 1, in <module>
    from .cudamat import *
  File "/home/ubgpu/github/cudamat/cudamat/cudamat.py", line 14, in <module>
    os.path.dirname(__file__) or os.path.curdir, 'libcudamat' + sysconfig.get_config_var('SO')))
  File "/usr/lib/python2.7/ctypes/__init__.py", line 443, in LoadLibrary
    return self._dlltype(name)
  File "/usr/lib/python2.7/ctypes/__init__.py", line 365, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /home/ubgpu/github/cudamat/cudamat/libcudamat.so: cannot open shared object file: No such file or directory

======================================================================
ERROR: Failure: ImportError (No module named ae)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/autoencoders/tests/test_ae.py", line 1, in <module>
    from ae import Autoencoder
ImportError: No module named ae

======================================================================
ERROR: Failure: ImportError (No module named ca)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/autoencoders/tests/test_cae.py", line 1, in <module>
    from ca import ContractiveAutoencoder
ImportError: No module named ca

======================================================================
ERROR: Failure: ImportError (No module named da)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/autoencoders/tests/test_dae.py", line 1, in <module>
    from da import DenoisingAutoencoder
ImportError: No module named da

======================================================================
ERROR: Failure: ImportError (No module named da)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/autoencoders/tests/test_dae_pento.py", line 1, in <module>
    from da import DenoisingAutoencoder
ImportError: No module named da

======================================================================
ERROR: Failure: ImportError (No module named pae)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/autoencoders/tests/test_pae.py", line 1, in <module>
    from pae import PowerupAutoencoder
ImportError: No module named pae

======================================================================
ERROR: Failure: ImportError (No module named pae)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/autoencoders/tests/test_pae_tfd.py", line 1, in <module>
    from pae import PowerupAutoencoder
ImportError: No module named pae

======================================================================
ERROR: Failure: ImportError (No module named sa)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/autoencoders/tests/test_sa.py", line 1, in <module>
    from sa import SparseAutoencoder
ImportError: No module named sa

======================================================================
ERROR: Failure: ImportError (No module named ipdb)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/cle/cle/cost/__init__.py", line 1, in <module>
    import ipdb
ImportError: No module named ipdb

======================================================================
ERROR: Failure: ImportError (No module named ipdb)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/cle/cle/data/__init__.py", line 1, in <module>
    import ipdb
ImportError: No module named ipdb

======================================================================
ERROR: Failure: OSError ([Errno 13] Permission denied: '/home/ubgpu/.theano/compiledir_Linux-3.16--generic-x86_64-with-Ubuntu-14.04-trusty-x86_64-2.7.6-64/tmp0Lt4HM')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/cle/cle/graph/__init__.py", line 1, in <module>
    import theano
  File "/usr/local/lib/python2.7/dist-packages/theano/__init__.py", line 98, in <module>
    theano.sandbox.cuda.tests.test_driver.test_nvidia_driver1()
  File "/usr/local/lib/python2.7/dist-packages/theano/sandbox/cuda/tests/test_driver.py", line 32, in test_nvidia_driver1
    profile=False)
  File "/usr/local/lib/python2.7/dist-packages/theano/compile/function.py", line 266, in function
    profile=profile)
  File "/usr/local/lib/python2.7/dist-packages/theano/compile/pfunc.py", line 511, in pfunc
    on_unused_input=on_unused_input)
  File "/usr/local/lib/python2.7/dist-packages/theano/compile/function_module.py", line 1465, in orig_function
    on_unused_input=on_unused_input).create(
  File "/usr/local/lib/python2.7/dist-packages/theano/compile/function_module.py", line 1103, in __init__
    theano.gof.cc.get_module_cache().refresh()
  File "/usr/local/lib/python2.7/dist-packages/theano/gof/cc.py", line 85, in get_module_cache
    return cmodule.get_module_cache(config.compiledir, init_args=init_args)
  File "/usr/local/lib/python2.7/dist-packages/theano/gof/cmodule.py", line 1390, in get_module_cache
    _module_cache = ModuleCache(dirname, **init_args)
  File "/usr/local/lib/python2.7/dist-packages/theano/gof/cmodule.py", line 617, in __init__
    self.refresh()
  File "/usr/local/lib/python2.7/dist-packages/theano/gof/cmodule.py", line 692, in refresh
    files = os.listdir(root)
OSError: [Errno 13] Permission denied: '/home/ubgpu/.theano/compiledir_Linux-3.16--generic-x86_64-with-Ubuntu-14.04-trusty-x86_64-2.7.6-64/tmp0Lt4HM'

======================================================================
ERROR: Failure: ImportError (No module named ipdb)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/cle/cle/layers/__init__.py", line 1, in <module>
    import ipdb
ImportError: No module named ipdb

======================================================================
ERROR: Failure: ImportError (No module named ipdb)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/cle/cle/models/__init__.py", line 1, in <module>
    import ipdb
ImportError: No module named ipdb

======================================================================
ERROR: Failure: ImportError (No module named ipdb)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/cle/cle/train/__init__.py", line 1, in <module>
    import ipdb
ImportError: No module named ipdb

======================================================================
ERROR: Failure: ImportError (No module named ipdb)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/cle/cle/utils/__init__.py", line 1, in <module>
    import ipdb
ImportError: No module named ipdb

======================================================================
ERROR: Failure: AttributeError ('module' object has no attribute 'tests')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/gensim/gensim/__init__.py", line 6, in <module>
    from gensim import parsing, matutils, interfaces, corpora, models, similarities
  File "/home/ubgpu/github/gensim/gensim/corpora/__init__.py", line 18, in <module>
    from .sharded_corpus import ShardedCorpus
  File "/home/ubgpu/github/gensim/gensim/corpora/sharded_corpus.py", line 34, in <module>
    import theano
  File "/usr/local/lib/python2.7/dist-packages/theano/__init__.py", line 83, in <module>
    if hasattr(theano.tests, "TheanoNoseTester"):
AttributeError: 'module' object has no attribute 'tests'
-------------------- >> begin captured logging << --------------------
gensim.utils: INFO: smart_open library not found; falling back to local-filesystem-only
gensim.utils: INFO: smart_open library not found; falling back to local-filesystem-only
--------------------- >> end captured logging << ---------------------

======================================================================
ERROR: Failure: AttributeError ('module' object has no attribute 'tests')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/gensim/gensim/__init__.py", line 6, in <module>
    from gensim import parsing, matutils, interfaces, corpora, models, similarities
  File "/home/ubgpu/github/gensim/gensim/corpora/__init__.py", line 18, in <module>
    from .sharded_corpus import ShardedCorpus
  File "/home/ubgpu/github/gensim/gensim/corpora/sharded_corpus.py", line 34, in <module>
    import theano
  File "/usr/local/lib/python2.7/dist-packages/theano/__init__.py", line 83, in <module>
    if hasattr(theano.tests, "TheanoNoseTester"):
AttributeError: 'module' object has no attribute 'tests'

======================================================================
ERROR: Failure: AttributeError ('module' object has no attribute 'tests')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/gensim/gensim/__init__.py", line 6, in <module>
    from gensim import parsing, matutils, interfaces, corpora, models, similarities
  File "/home/ubgpu/github/gensim/gensim/corpora/__init__.py", line 18, in <module>
    from .sharded_corpus import ShardedCorpus
  File "/home/ubgpu/github/gensim/gensim/corpora/sharded_corpus.py", line 34, in <module>
    import theano
  File "/usr/local/lib/python2.7/dist-packages/theano/__init__.py", line 83, in <module>
    if hasattr(theano.tests, "TheanoNoseTester"):
AttributeError: 'module' object has no attribute 'tests'

======================================================================
ERROR: Failure: AttributeError ('module' object has no attribute 'tests')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/gensim/gensim/__init__.py", line 6, in <module>
    from gensim import parsing, matutils, interfaces, corpora, models, similarities
  File "/home/ubgpu/github/gensim/gensim/corpora/__init__.py", line 18, in <module>
    from .sharded_corpus import ShardedCorpus
  File "/home/ubgpu/github/gensim/gensim/corpora/sharded_corpus.py", line 34, in <module>
    import theano
  File "/usr/local/lib/python2.7/dist-packages/theano/__init__.py", line 83, in <module>
    if hasattr(theano.tests, "TheanoNoseTester"):
AttributeError: 'module' object has no attribute 'tests'

======================================================================
ERROR: Failure: AttributeError ('module' object has no attribute 'tests')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/gensim/gensim/__init__.py", line 6, in <module>
    from gensim import parsing, matutils, interfaces, corpora, models, similarities
  File "/home/ubgpu/github/gensim/gensim/corpora/__init__.py", line 18, in <module>
    from .sharded_corpus import ShardedCorpus
  File "/home/ubgpu/github/gensim/gensim/corpora/sharded_corpus.py", line 34, in <module>
    import theano
  File "/usr/local/lib/python2.7/dist-packages/theano/__init__.py", line 83, in <module>
    if hasattr(theano.tests, "TheanoNoseTester"):
AttributeError: 'module' object has no attribute 'tests'

======================================================================
ERROR: Failure: AttributeError ('module' object has no attribute 'tests')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/gensim/gensim/__init__.py", line 6, in <module>
    from gensim import parsing, matutils, interfaces, corpora, models, similarities
  File "/home/ubgpu/github/gensim/gensim/corpora/__init__.py", line 18, in <module>
    from .sharded_corpus import ShardedCorpus
  File "/home/ubgpu/github/gensim/gensim/corpora/sharded_corpus.py", line 34, in <module>
    import theano
  File "/usr/local/lib/python2.7/dist-packages/theano/__init__.py", line 83, in <module>
    if hasattr(theano.tests, "TheanoNoseTester"):
AttributeError: 'module' object has no attribute 'tests'

======================================================================
ERROR: Failure: AttributeError ('module' object has no attribute 'tests')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 411, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/ubgpu/github/gensim/gensim/__init__.py", line 6, in <module>
    from gensim import parsing, matutils, interfaces, corpora, models, similarities
  File "/home/ubgpu/github/gensim/gensim/corpora/__init__.py", line 18, in <module>
    from .sharded_corpus import ShardedCorpus
  File "/home/ubgpu/github/gensim/gensim/corpora/sharded_corpus.py", line 34, in <module>
    import theano
  File "/usr/local/lib/python2.7/dist-packages/theano/__init__.py", line 83, in <module>
    if hasattr(theano.tests, "TheanoNoseTester"):
AttributeError: 'module' object has no attribute 'tests'

======================================================================
FAIL: test_ggmm.test_time_do_mstep(1000, 16, 200)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/ubgpu/github/ggmm/tests/test_ggmm.py", line 384, in check_time_do_mstep
    assert_greater(speedup, 1.0)
AssertionError: 0.7825025736392919 not greater than 1.0
-------------------- >> begin captured stdout << ---------------------

------------------------------------
N=1000, K=16, D=200
avg_cpu_time (10 trials): 0.00112359523773
avg_gpu_time (10 trials): 0.0014358997345
speedup: 0.782502573639
------------------------------------


--------------------- >> end captured stdout << ----------------------

----------------------------------------------------------------------
Ran 72 tests in 8.195s

FAILED (errors=24, failures=1)
ubgpu@ubgpu:~/github$

greate.


ubgpu@ubgpu:~/github/cudamat/test$ nosetests
..................................F.....................E
======================================================================
ERROR: test suite for <module 'test_learn' from '/home/ubgpu/github/cudamat/test/test_learn.pyc'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/suite.py", line 208, in run
    self.setUp()
  File "/usr/lib/python2.7/dist-packages/nose/suite.py", line 291, in setUp
    self.setupContext(ancestor)
  File "/usr/lib/python2.7/dist-packages/nose/suite.py", line 314, in setupContext
    try_run(context, names)
  File "/usr/lib/python2.7/dist-packages/nose/util.py", line 471, in try_run
    return func()
  File "/home/ubgpu/github/cudamat/test/test_learn.py", line 7, in setup
    cm.cublas_init()
  File "/home/ubgpu/.local/lib/python2.7/site-packages/cudamat-0.3-py2.7-linux-x86_64.egg/cudamat/cudamat.py", line 1502, in cublas_init
    raise CUDAMatException('error initializing CUBLAS: (err=%u)' % err)
CUDAMatException: error initializing CUBLAS: (err=-2)

======================================================================
FAIL: test_cudamat.test_pow_matrix
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/ubgpu/github/cudamat/test/test_cudamat.py", line 791, in test_pow_matrix
    assert np.max(np.abs(c - m1.numpy_array)) < 10**-2, "Error in cudamat.pow exceeded threshold"
AssertionError: Error in cudamat.pow exceeded threshold

----------------------------------------------------------------------
Ran 56 tests in 3.660s

FAILED (errors=1, failures=1)
ubgpu@ubgpu:~/github/cudamat/test$

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -45,8 +45,9 @@ Testing
 To test your setup, run the included unit tests and optionally the benchmark:
 
 ```bash
+cd test  # so it doesn't try importing cudamat from the source directory
 # Run tests
 nosetests
 # Run benchmark
-python examples/bench_cudamat.py
+python ../examples/bench_cudamat.py
 ```

